src/include/sof/drivers/timer.h

```diff
@@ -76,6 +77,16 @@ static inline uint64_t platform_safe_get_time(struct timer *timer)
 void platform_timer_start(struct timer *timer);
 void platform_timer_stop(struct timer *timer);
 
+static inline uint64_t platform_timer_timeout_calc_ms(struct timer *timer, uint64_t ms)
+{
+	return platform_timer_get(timer) + clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, ms);
+}
+
+static inline uint64_t platform_timer_timeout_calc_us(struct timer *timer, uint64_t us)
+{
+	return platform_timer_get(timer) + clock_us_to_ticks(PLATFORM_DEFAULT_CLOCK, us);
+}
```

Comment:
in Zephyr `timeout` has a specific meaning: see the `k_timeout_t` type. Since SOF will change to using Zephyr APIs natively, wouldn't the use of "timeout" in these two functions cause confusion? What they do they just add a certain time to "now," right? So, maybe just `platform_timer_add_us()` and `..._ms()`?

Reply:
I changed every timeout/deadline calculation from

timeout = platform_timer_get(timer) + clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1);

to:

timeout = platform_timer_timeout_calc_ms(timer, 1);

which will use sys_clock_timeout_end_calc on Zephyr.